### PR TITLE
Improve documention

### PR DESCRIPTION
**Pull request by MaceM8**
Add more thorough documentation with more complete stories for each public component. 
Added MDX support to storybook and included addon-docs as well.

### CODE REVIEW

File: .plop/uiComponent/statelessComponent.js.hbs [hunk 1]
```diff
@@ -2,17 +2,17 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { cx } from 'ramda-extension';
 
-const {{ pascalCase name }} = ({ children, className, ...otherProps }) => (
-	<div className={cx(className)} {...otherProps}>
-		{children}
-	</div>
+const {{ pascalCase name }} = ({ children, className, ...rest }) => (
+<div className={cx(className)} {...rest}>
```
Comment (tommmyy) [inline .plop/uiComponent/statelessComponent.js.hbs]: missing indentation

File: apps/storybook/.storybook/util/sortStories.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Inspired by: https://github.com/sumup-oss/circuit-ui/blob/598f9630ff049b0768485670d6e6f978660e4072/.storybook/util/story-helpers.js
+const splitStoryName = name => name.split('/');
```
Comment (tommmyy) [inline apps/storybook/.storybook/util/sortStories.js]: import { split } from 'ramda'; 
const splitStoryName = split('/');

File: apps/storybook/.storybook/util/sortStories.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+const position = (item, array = []) => {
```
Comment (tommmyy) [inline apps/storybook/.storybook/util/sortStories.js]: https://ramda-extension.firebaseapp.com/repl/?v=%5E0.26.1&ve=0.10.1#?const%20getPosition%20%3D%20R_.composeC%28%0A%20%20when%28equals%28-1%29%2C%20always%281000%29%29%2C%0A%20%20indexOf%0A%29%3B%0A%0AgetPosition%28%22a%22%2C%20%5B%22b%22%2C%20%22a%22%5D%29%0AgetPosition%28%22a%22%2C%20%5B%22a%22%5D%29

better rename to "getSortPosition"

File: packages/lundium/src/components/FormControl/FormControl.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import React, { createRef, useContext, useEffect } from 'react';
```
Comment (tommmyy) [inline packages/lundium/src/components/FormControl/FormControl.js:1]: ❤️

File: packages/lundium/src/components/FormControl/FormControl.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	size,
+	type,
+	value,
+	...rest
+}) => {
+	const { onFilled, onEmpty, hasError, hasFloatingLabel, onFocus, onBlur } = useContext(
+		FormGroupContext
+	);
+
+	useEffect(() => {
```
Comment (tommmyy) [inline packages/lundium/src/components/FormControl/FormControl.js:34]: these two hooks can be merged into one

File: packages/lundium/src/components/FormControl/FormControl.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	onFocus: PropTypes.func,
+	/** Size of the form control. */
+	size: PropTypes.oneOf(['sm', 'lg']),
+	/** Type attribute of the form control. */
+	type: PropTypes.string,
+	/** Value of the form control. */
+	value: PropTypes.any,
+};
+
+FormControl.defaultProps = {
```
Comment (tommmyy) [inline packages/lundium/src/components/FormControl/FormControl.js]: remove defaultProps and use default args

File: packages/lundium/src/components/FormControl/FormControl.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+		onFocus();
+		return rest.onFocus(e);
+	};
+
+	const handleBlur = e => {
+		onBlur();
+		return rest.onBlur(e);
+	};
+
+	const isCheckable = ['radio', 'checkbox'].includes(type) || isNotNil(checked);
```
Comment (tommmyy) [inline packages/lundium/src/components/FormControl/FormControl.js]: `['radio', 'checkbox'].includes` 
can be rewirtten:

```
import { flipIncludes } frrom "ramda-extension"

const isTypeCheckable = flipIncludes(['radio', 'checkbox']);

const FormControl = () => {
  // ...

  const isCheckable = isTypeCheckable || isNotNil(checked);
}
```

File: packages/lundium/src/components/FormField/FormField.js [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+	required: PropTypes.bool,
+	/** Assign a suffix component (e.g. Icon) to Input */
+	suffix: PropTypes.node,
+	/** Node shown in warning state. */
+	warning: PropTypes.node,
+};
+
+export { FormField };
+
+export default withGeneratedID(FormField);
```
Comment (tommmyy) [inline packages/lundium/src/components/FormField/FormField.js]: hook can be used istead - useGeneratedId

File: packages/lundium/src/components/InlineText/InlineText.js [hunk 9]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types';
 
 import Text from '../Text';
 
-const InlineText = ({ as = 'span', ...props }) => <Text {...props} as={as} />;
+const InlineText = ({ as = 'span', ...rest }) => <Text {...rest} as={as} />;
```
Comment (tommmyy) [inline packages/lundium/src/components/InlineText/InlineText.js]: my bad... "as" should be before `...rest`

File: packages/lundium/src/components/Label/Label.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { cx } from 'ramda-extension';
+
+// import { useTheme } from '../ThemeProvider';
+
+const Label = ({ children, className, isHidden, disabled, readOnly, ...rest }) => {
+	// const theme = useTheme();
```
Comment (tommmyy) [inline packages/lundium/src/components/Label/Label.js]: ?

File: packages/lundium/src/utils/withGeneratedID.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import React, { forwardRef } from 'react';
```
Comment (tommmyy) [inline packages/lundium/src/utils/withGeneratedID.js]: can be removed. hook is sufficient

File: packages/theme-basic/src/scss/components/FormGroup.scss [hunk 12]
```diff
@@ -0,0 +1,287 @@
...
+
+		[class^='icon'],
+		[class*='icon'] {
+			// stylelint-disable-next-line
+			color: $primary !important;
+		}
+	}
+}
+
+.password-strength {
```
Comment (tommmyy) [inline packages/theme-basic/src/scss/components/FormGroup.scss]: remove

**Review by tommmyy — CHANGES_REQUESTED** [review]
Awesome work! I have just few minor things.

File: packages/lundium/src/components/Label/Label.js (hunk 10 above)
Comment (MaceM8) [reply packages/lundium/src/components/Label/Label.js]: My bad, missed this one

**Pull request merged**